VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/MissingBraces/MissingBracesAnalyzer.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.MissingBraces
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class MissingBracesAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.MissingBracesAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.MissingBracesAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            => new DiagnosticDescriptor(DiagnosticId.MissingBraces, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeSymbol,
```

Comment:
You could just register separate `SyntaxNodeActions` -- then you don't have to do the iskind + cast check anymore which would greatly improve performance I think